clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DeployCloudFoundryServerGroupAtomicOperationTest.java

```diff
@@ -66,8 +66,114 @@ void convertToMbHandling() {
+    final DeployCloudFoundryServerGroupAtomicOperation operation =
+      new DeployCloudFoundryServerGroupAtomicOperation(new PassThroughOperationPoller(), description, clusterProvider);
+    final Applications apps = getApplications(clusterProvider, ProcessStats.State.CRASHED);
+
+    // When
+    final DeploymentResult result = operation.operate(Lists.emptyList());
+
+    // Then
+    final InOrder inOrder = Mockito.inOrder(apps, cloudFoundryClient.getServiceInstances());
+    inOrder.verify(apps).createApplication("app1-stack1-detail1-v000",
+      CloudFoundrySpace.builder().id("space1Id").name("space1").build(),
+      io.vavr.collection.List.of("buildpack1", "buildpack2").asJava(),
+      HashMap.of(
+        "token", "ASDF"
+      ).toJavaMap());
+    inOrder.verify(apps).uploadPackageBits(eq("serverGroupId_package"), any());
+    inOrder.verify(apps).createBuild("serverGroupId_package");
+    inOrder.verify(apps).scaleApplication("serverGroupId", 7, 1024, 2048);
+    inOrder.verify(apps).updateProcess("serverGroupId", null, "http", "/health");
+    inOrder.verify(cloudFoundryClient.getServiceInstances()).createServiceBindingsByName(any(), eq(Collections.singletonList("service1")));
+    inOrder.verify(apps).startApplication("serverGroupId");
+
+    assertThat(testTask.getStatus().isFailed()).isTrue();
+  }
+
+  @Test
+  void executeOperationWithNoStartFlag() {
+    // Given
+    final DeployCloudFoundryServerGroupDescription description = getDeployCloudFoundryServerGroupDescription(false);
+    final CloudFoundryClusterProvider clusterProvider = mock(CloudFoundryClusterProvider.class);
+    final DeployCloudFoundryServerGroupAtomicOperation operation =
+      new DeployCloudFoundryServerGroupAtomicOperation(new PassThroughOperationPoller(), description, clusterProvider);
+    final Applications apps = getApplications(clusterProvider, ProcessStats.State.RUNNING);
+
+    // When
+    final DeploymentResult result = operation.operate(Lists.emptyList());
+
+    // Then
+    final InOrder inOrder = Mockito.inOrder(apps, cloudFoundryClient.getServiceInstances());
+    inOrder.verify(apps).createApplication("app1-stack1-detail1-v000",
+      CloudFoundrySpace.builder().id("space1Id").name("space1").build(),
+      io.vavr.collection.List.of("buildpack1", "buildpack2").asJava(),
+      HashMap.of(
+        "token", "ASDF"
+      ).toJavaMap());
+    inOrder.verify(apps).uploadPackageBits(eq("serverGroupId_package"), any());
+    inOrder.verify(apps).createBuild("serverGroupId_package");
+    inOrder.verify(apps).scaleApplication("serverGroupId", 7, 1024, 2048);
+    inOrder.verify(apps).updateProcess("serverGroupId", null, "http", "/health");
+    inOrder.verify(cloudFoundryClient.getServiceInstances()).createServiceBindingsByName(any(), eq(Collections.singletonList("service1")));
+    inOrder.verify(apps, never()).startApplication("serverGroupId");
```

Comment:
This is the important part. It gets lost in all the other verifications which are mostly duplicated between the two tests. Is there a way to set this important verify apart and combine the common ones?

Reply:
I'm sure we can figure that out. I'll take a look at it.

Reply:
I think this solves the issue, @claymccoy please take a look.